### PR TITLE
fix the CRLF issue, review comments related to allow/deny user/groups

### PR DESCRIPTION
**Pull request by bagajjal**
1) https://github.com/PowerShell/Win32-OpenSSH/issues/861

2) Fixing review comments related to allow/deny user/group

### CODE REVIEW

File: contrib/win32/win32compat/misc.c [hunk 1]
```diff
@@ -1116,27 +1120,28 @@ getusergroups(const char *user, int *ngroups)
 		*t='\0';
 	}
 
-	w_username = utf8_to_utf16(user_name);
-	if (!w_username) {
+	user_name_utf16 = utf8_to_utf16(user_name);
```
Comment (bingbing8) [inline contrib/win32/win32compat/misc.c:1126]: user_name_utf16  needs to be free

File: contrib/win32/win32compat/misc.c [hunk 2]
```diff
@@ -1116,27 +1120,28 @@ getusergroups(const char *user, int *ngroups)
...
-	if (get_machine_domain_name(w_machine_domain_name, 64)) {
+	wchar_t *machine_domain_name_utf16 = NULL;
+	get_machine_domain_name(&machine_domain_name_utf16);
+	if (machine_domain_name_utf16) {
 		if(!machine_domain_name)
-			machine_domain_name = utf16_to_utf8(w_machine_domain_name);
+			machine_domain_name = utf16_to_utf8(machine_domain_name_utf16);
 
-		wchar_t local_user_fmt[256] = { 0, };
+		wchar_t local_user_fmt_utf16[256] = { 0, };
```
Comment (bingbing8) [inline contrib/win32/win32compat/misc.c]: again, can we use UNCLEN + DNLEN + 2?

**Review by bingbing8 — CHANGES_REQUESTED** [review]

**Comment by DHowett-MSFT** [comment]
This pull request seems to be two different pull requests stuck together. The user/group changes should not be causing the CRLF breakage in Nano, right?

**Comment by bagajjal** [comment]
@DHowett-MSFT - Yes, user/group related changes are the review comments from another PR... The original PR related to user/group is closed so I am fixing the  user/group related changes here... I have updated the description to make it explicit.

**Review by bingbing8 — APPROVED** [review]

**Pull request merged**